clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -976,15 +1053,14 @@ public void subscribe(Collection<String> topics) {
      * Group rebalances only take place during an active call to {@link #poll(Duration)}.
      *
      * @param pattern Pattern to subscribe to
-     * @param listener Non-null listener instance to get notifications on partition assignment/revocation for the
-     *                 subscribed topics
+     * @param listener {@link Optional} listener instance to get notifications on partition assignment/revocation
+     *                 for the subscribed topics
      * @throws IllegalArgumentException If pattern or listener is null
      * @throws IllegalStateException If {@code subscribe()} is called previously with topics, or assign is called
      *                               previously (without a subsequent call to {@link #unsubscribe()}), or if not
      *                               configured at-least one partition assignment strategy
      */
-    @Override
-    public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
+    private void subscribe(Pattern pattern, Optional<ConsumerRebalanceListener> listener) {
         maybeThrowInvalidGroupIdException();
         if (pattern == null || pattern.toString().equals(""))
```

Comment:
it is a bit weird to use toString() instead of `pattern()` even though both implementation return the same field, `pattern`

Reply:
Agreed. I'm loath to change it at this point due to some weird side effect of which I'm unaware.